### DISCUSSION TITLE
Sentences that start in a line and end in the other line (PROBLEM) - Any solution or post processing tool for that?

### DISCUSSION THREAD

**Discussion by NormalFall0** [openai/whisper#2182] [Q&A]
Hi,
Sometimes a sentence is started in a line and is continued until the next line, for example:

> "
> Timestamp1
> Hello everyrone, today were are going to show
> 
> Timestamp2
> you ten great recipes from Australia. Let's start with
> 
> Timestamp3
> ...
> "

I was there was a way to have full sentences within one line only.
So the timestamp for the srt, would include ALL the duration one ONE FULL SENTENCE, in one line/ one timestamp.

So it should be like this:

> "
> Timestamp1 (modified)
> Hello everyone, today were are going to show you ten great recipes from Australia
> 
> Timestamp2 (modified)
> Let's start with ..
> 
> Timestamp3
> ...
> "

What solutions are there for this problem? Maybe I am not aware of existing tools/techs that could help somehow?

Thanks

**Comment by whicks1** [comment]
A: Use the python webvtt library to parse a saved vtt or srt output and loop over the cues. Keep track of periods and split/generate new cues based on the sentences. It’ll get weird with abbreviations like Mr. Dr. Etc so you’ll need a map of those to ignore. And you may want to recalculate a start/end timestamp for cues and figure out an avg # of words per cue you are dealing with. Outputs are going to be close but not perfect. You’ll also might run into  oddities like the occasional casing/punctuation loss which would affect your output along with odd hallucinations loops. 

B. Probably better option: use the JSON output with word timestamps and build your srt or vtt on your own.
